lib/vsc/install/ci.py

```diff
@@ -62,19 +65,70 @@
 PY3_ONLY = 'py3_only'
 PY3_TESTS_MUST_PASS = 'py3_tests_must_pass'
 RUN_SHELLCHECK = 'run_shellcheck'
+ENABLE_GITHUB_ACTIONS = 'enable_github_actions'
 
 logging.basicConfig(format="%(message)s", level=logging.INFO)
 
 
 def write_file(path, txt):
     """Write specified contents to specified path."""
+
+    if not os.path.exists(os.path.dirname(path)):
+        os.makedirs(os.path.dirname(path))
     try:
         with open(path, 'w') as handle:
             handle.write(txt)
         logging.info("Wrote %s", path)
     except (IOError, OSError) as err:
         raise IOError("Failed to write %s: %s" % (path, err))
 
+def gen_github_action(repo_base_dir=os.getcwd()):
+    """
+    Generate tox.ini configuration file for github actions.
+    """
+    logging.info('[%s]', GITHUB_ACTIONS)
+    vsc_ci_cfg = parse_vsc_ci_cfg()
+
+    setup = vsc_setup()
+    repofile = os.path.join(repo_base_dir, ".git/config")
+    name_url = setup.get_name_url(filename=repofile, version='ALL_VERSIONS')['url']
+
+    if vsc_ci_cfg[ENABLE_GITHUB_ACTIONS]:
+        header = [
+            "%s: configuration file for github actions worflow" % GITHUB_ACTIONS,
+            "This file was automatically generated using 'python -m vsc.install.ci'",
+            "DO NOT EDIT MANUALLY",
+        ]
+
+        txt = ['# ' + l for l in header]
+        yaml_content = {
+            'name': 'run python tests',
+            'on': ['push', 'pull_request'],
+            'jobs': {
+                'python_unittests': {
+                    'runs-on': 'ubuntu-20.04',
+                    'strategy': {
+                        'matrix': {
+                            'python': [3.6, 3.9]
+                        }
+                    },
+                    'steps': [
+                        {'name': 'Checkout code', 'uses': 'actions/checkout@v3'},
+                        {'name': 'Setup Python', 'uses': 'actions/setup-python@v4',
+                         'with': {'python-version': '${{ matrix.python }}'}},
+                        {'name': 'install tox', 'run': 'pip install tox'},
+                        {'name': 'add mandatory git remote',
+                         'run': 'git remote add hpcugent %s.git' % name_url},
```

Comment:
ue f string

Reply:
I will, as soon as this package is python 3 only :wink: